vllm/utils.py

```diff
@@ -693,3 +693,38 @@ def inner(*args, **kwargs):
         return inner  # type: ignore
 
     return wrapper
+
+
+@lru_cache(maxsize=8)
```

Comment:
cuda_visible_devices can have more than 8 possible values :)

Reply:
this will cache based on CUDA_VISIBLE_DEVICES input argument value, which generally should only be set twice (at the start and when initializing workers).

Reply:
I understand CUDA_VISIBLE_DEVICES is set for several times in general. But I don't feel we should put a magic number here. Set the size to None will not cost infinite resource. It is dynamic (and can even save resource for you, because you typically don't set it for 8 times).

Reply:
Sorry, I don't follow. Setting it to None will definitely not save any resources - the cache is not preallocated (it's just backed by a dict). The size of the cache is purely an implementation detail that will not change the logic. Indeed, we could just get rid of the cache altogether, if we were not concerned about performance. I disagree as a matter of principle for allowing any cache to grow infinitely, even if the chance of that is small.

Reply:
please go ahead then 👍